src/app/crypto/services/pgp.service.ts

```diff
@@ -15,53 +19,95 @@ export async function generateNewKeys(): Promise<{
     curve: 'ed25519',
   });
 
+  const kem = await kemBuilder();
+  const { publicKey: publicKyberKey, privateKey: privateKyberKey } = await kem.keypair();
+
   return {
     privateKeyArmored: privateKey,
     publicKeyArmored: Buffer.from(publicKey).toString('base64'),
     revocationCertificate: Buffer.from(revocationCertificate).toString('base64'),
+    publicKyberKeyBase64: Buffer.from(publicKyberKey).toString('base64'),
+    privateKyberKeyBase64: Buffer.from(privateKyberKey).toString('base64'),
   };
 }
+export function XORhex(a: string, b: string): string {
+  let res = '',
+    i = a.length,
+    j = b.length;
+  while (i-- > 0 && j-- > 0) res = (parseInt(a.charAt(i), 16) ^ parseInt(b.charAt(j), 16)).toString(16) + res;
+  return res;
+}
 
 export const encryptMessageWithPublicKey = async ({
   message,
   publicKeyInBase64,
+  publicKyberKeyBase64,
 }: {
   message: string;
   publicKeyInBase64: string;
-}): Promise<WebStream<string>> => {
+  publicKyberKeyBase64: string;
+}): Promise<string> => {
   const openpgp = await getOpenpgp();
+  const kem = await kemBuilder();
 
   const publicKeyArmored = Buffer.from(publicKeyInBase64, 'base64').toString();
   const publicKey = await openpgp.readKey({ armoredKey: publicKeyArmored });
 
+  const publicKyberKey = Buffer.from(publicKyberKeyBase64, 'base64');
+  const { ciphertext, sharedSecret: secret } = await kem.encapsulate(publicKyberKey);
+  const kyberCiphertextStr = Buffer.from(ciphertext).toString('base64');
+  const secretHex = Buffer.from(secret).toString('hex');
+  const messageHex = Buffer.from(message).toString('hex');
+
+  // message should be the same length as secret, which is 256 bits
+  const xoredMessage = XORhex(messageHex, secretHex);
```

Comment:
I think that a custom error in which it is indicated that the length is not equal and that it should be. Provided that it has to be the case of course, that if the length is wrong, it will give problems. 
This way the error is already managed in the place where the function is called, so that there are no inconsistencies and problems with the encryption, before encrypting something wrong, better that it gives an error and warns to be able to see why the length of the strings is different